Source/Core/DolphinQt/Host.cpp

```diff
@@ -154,3 +157,12 @@ bool Host_UINeedsControllerState()
 void Host_RefreshDSPDebuggerWindow()
 {
 }
+
+void Host_PPCTitleChanged()
+{
+#ifdef USE_DISCORD_PRESENCE
+  // TODO: Not sure if the NetPlay check is needed.
+  if (!NetPlay::IsNetPlayRunning())
```

Comment:
Any particular reason why NetPlay shouldn't update the Discord presence?

Reply:
MainWindow::StartGame() checks for netplay before updating Discord presence:
https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/DolphinQt/MainWindow.cpp#L916-L919

I'm not sure if there's a particular reason for doing so.

Reply:
Ah, it seems to be because netplay does it on its own, with some extra data: https://github.com/dolphin-emu/dolphin/blob/f0ac74d1b05a124023c37db202c6fa73a3ee3aad/Source/Core/DolphinQt/NetPlay/NetPlayDialog.cpp#L574

So the correct way to handle this isn't to just skip updating the Discord presence if we're running netplay, it's to update the Discord presence and include the extra netplay data.

Reply:
Maybe @yourWaifu can shed some light on this; altho it might be related to Qt calling it from the wrong thread.

Reply:
I can't exactly remember why I did it but I think it has to do with Netplay has it's own UpdateDiscordPresence to add data that it needs for join request. Once the game starts, netplay will use it's own UpdateDiscordPresence to send an update without the netplay join data to prevent others from asking to join while the game is running. Without the if, it'll send an extra update to Discord using Discord::UpdateDiscordPresence and NetPlayDialog::UpdateDiscordPresence. You don't really need the if, removing isn't noticeable to the user. However, it might be a good idea to send less updates to prevent hitting rate limits or save bandwidth or something. I don't think it's a huge deal really but I would recommended there being a check if Netplay will cause a 2nd update being sent. In this case, I'm not really sure if there's a 2nd update being sent when the PPC title changes. So, I'll do some testing on that.